pkg/client/optimizely_user_context.go

```diff
@@ -0,0 +1,109 @@
+/****************************************************************************
+ * Copyright 2020, Optimizely, Inc. and contributors                        *
+ *                                                                          *
+ * Licensed under the Apache License, Version 2.0 (the "License");          *
+ * you may not use this file except in compliance with the License.         *
+ * You may obtain a copy of the License at                                  *
+ *                                                                          *
+ *    http://www.apache.org/licenses/LICENSE-2.0                            *
+ *                                                                          *
+ * Unless required by applicable law or agreed to in writing, software      *
+ * distributed under the License is distributed on an "AS IS" BASIS,        *
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. *
+ * See the License for the specific language governing permissions and      *
+ * limitations under the License.                                           *
+ ***************************************************************************/
+
+// Package client //
+package client
+
+import (
+	"sync"
+
+	"github.com/optimizely/go-sdk/pkg/decide"
+	"github.com/optimizely/go-sdk/pkg/entities"
+)
+
+// OptimizelyUserContext defines user contexts that the SDK will use to make decisions for.
+type OptimizelyUserContext struct {
+	userID     string
+	attributes map[string]interface{}
+	optimizely *OptimizelyClient
+	mutex      *sync.RWMutex
+}
+
+// returns an instance of the optimizely user context.
+func newOptimizelyUserContext(optimizely *OptimizelyClient, userID string, attributes map[string]interface{}) OptimizelyUserContext {
```

Comment:
it could take old context instead of userId and attributes

Reply:
I recommend not to support the old context here. It can be a bit less convenient but will be much better to keep it clean from old context.